common/src/core/java/xyz/wagyourtail/jsmacros/core/library/impl/FJsMacros.java

```diff
@@ -446,18 +461,63 @@ public boolean off(String event, IEventListener listener) {
     }
 
     /**
+     * Will also disable all listeners for the given event, including JsMacros own event listeners.
+     *
      * @param event the event to remove all listeners from
      * @since 1.8.4
      */
-    public void disableListeners(String event) {
-        Core.getInstance().eventRegistry.getListeners().get(event).clear();
+    public void disableAllListeners(String event) {
+        for (IEventListener listener : ImmutableList.copyOf(Core.getInstance().eventRegistry.getListeners(event))) {
+            listener.off();
+        }
     }
 
     /**
+     * Will also disable all listeners, including JsMacros own event listeners.
+     *
      * @since 1.8.4
      */
     public void disableAllListeners() {
-        Core.getInstance().eventRegistry.getListeners().clear();
+        for (Map.Entry<String, Set<IEventListener>> entry : Core.getInstance().eventRegistry.getListeners().entrySet()) {
+            for (IEventListener listener : ImmutableList.copyOf(entry.getValue())) {
+                listener.off();
+            }
+        }
+    }
+
+    /**
+     * Will only disable user created event listeners for the given event. This includes listeners
+     * created from {@link #on(String, MethodWrapper)}, {@link #once(String, MethodWrapper)},
+     * {@link #waitForEvent(String)}, {@link #waitForEvent(String, MethodWrapper)} and
+     * {@link #waitForEvent(String, MethodWrapper, MethodWrapper)}.
+     *
+     * @param event the event to remove all listeners from
+     * @since 1.8.4
+     */
+    public void disableUserListeners(String event) {
+        for (IEventListener listener : ImmutableList.copyOf(Core.getInstance().eventRegistry.getListeners(event))) {
+            if (listener instanceof ScriptEventListener) {
+                listener.off();
+            }
+        }
+    }
+
+    /**
+     * Will only disable user created event listeners.  This includes listeners created from
+     * {@link #on(String, MethodWrapper)}, {@link #once(String, MethodWrapper)},
+     * {@link #waitForEvent(String)}, {@link #waitForEvent(String, MethodWrapper)} and
+     * {@link #waitForEvent(String, MethodWrapper, MethodWrapper)}.
+     *
+     * @since 1.8.4
+     */
+    public void disableUserListeners() {
+        for (Map.Entry<String, Set<IEventListener>> entry : Core.getInstance().eventRegistry.getListeners().entrySet()) {
+            for (IEventListener listener : ImmutableList.copyOf(entry.getValue())) {
+                if (listener instanceof ScriptEventListener) {
+                    listener.off();
+                }
+            }
+        }
```

Comment:
I think that the better name for this is `disableScriptListeners`, ~~also will this break `waitForEvent`, we don't want those threads to just be stuck in limbo when someone does this... should get them to throw an interrupt somehow~~